docs/install-components/karmada.md

```diff
@@ -0,0 +1,178 @@
+cd kurator
+hack/local-dev-setup.sh
+```
+
+### KinD
+
+install docker
+```bash
+sudo apt-get update
+sudo apt-get install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gnupg-agent \
+    software-properties-common -y
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo apt-key fingerprint 0EBFCD88
+
+sudo add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) \
+    stable"
+
+sudo apt-get update
+sudo apt-get install docker-ce docker-ce-cli containerd.io -y
+```
+
+Install KinD
+```bash
+curl -Lo ./kind "https://kind.sigs.k8s.io/dl/v0.11.1/kind-$(uname)-amd64"
+chmod +x ./kind
+mv ./kind /usr/local/bin/kind  
+```
+
+Create a kubernetes cluster
+```bash
+kind create cluster
+```
+
+For more instructions, please refer to [KinD](https://kind.sigs.k8s.io/docs/user/quick-start)
+
+### kubeadm
+
+Load Linux kernel modules and set kernel parameters.
+```bash
+cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
+overlay
+br_netfilter
+EOF
+
+sudo modprobe overlay
+sudo modprobe br_netfilter
+
+# sysctl params required by setup, params persist across reboots
+cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
+net.bridge.bridge-nf-call-iptables  = 1
+net.bridge.bridge-nf-call-ip6tables = 1
+net.ipv4.ip_forward                 = 1
+EOF
+
+# Apply sysctl params without reboot
+sudo sysctl --system
+```
+
+Install containerd
+```
+sudo apt-get install  containerd
+```
+
+Install Kubeadm Kubelet Kubectl
+```bash
+sudo apt-get update && sudo apt-get install -y apt-transport-https curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+cat <<EOF | sudo tee /etc/apt/sources.list.d/kubernetes.list
+deb https://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+
+sudo apt-get update
+sudo apt-get install -y kubelet kubeadm kubectl
+sudo apt-mark hold kubelet kubeadm kubectl
+```
+
+Create a kubernetes cluster
+```
+kubeadm config images pull
+kubeadm init
+
+mkdir -p $HOME/.kube
+sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u):$(id -g) $HOME/.kube/config
+```
+
+Install CNI, take calico as an example
+```
+curl https://docs.projectcalico.org/manifests/calico.yaml -O
+kubectl apply -f calico.yaml
+kubectl -n kube-system get pod -w
+```
+> Notice
+> 
+> In the kubernetes cluster created by Kubeadm, the master node is identified as Taints. If your kubernetes cluster has only one node, you need to remove the Taints of the master node.
+> 
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/control-plane-`
+> 
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/master-`
+
+
+For more instructions, please refer to [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm)
+
+## Deploy Karmada
+
+Compile `kurator` from source
+```bash
+git clone https://github.com/kurator-dev/kurator.git
+make kurator
+```
+
+Download `kurator` from the Releases page
```

Comment:
should we just remove this?

Reply:
I personally think that kurator will release binaries of Releases version in the future, we can't let each user compile kurator by himself.

Do you mean to wait for the release version to add it?

Reply:
> Do you mean to wait for the release version to add it?

yes

Reply:
done